Lib/asyncio/proactor_events.py

```diff
@@ -160,22 +160,26 @@ def __init__(self, loop, sock, protocol, waiter=None,
 
     def pause_reading(self):
         if self._closing:
-            raise RuntimeError('Cannot pause_reading() when closing')
-        if self._paused:
-            raise RuntimeError('Already paused')
-        self._paused = True
-        if self._loop.get_debug():
-            logger.debug("%r pauses reading", self)
+            return
+        elif self._paused:
+            if self._loop.get_debug():
+                logger.debug("%r already paused", self)
+        else:
+            self._paused = True
+            if self._loop.get_debug():
+                logger.debug("%r pauses reading", self)
 
     def resume_reading(self):
-        if not self._paused:
-            raise RuntimeError('Not paused')
-        self._paused = False
         if self._closing:
             return
-        self._loop.call_soon(self._loop_reading, self._read_fut)
-        if self._loop.get_debug():
-            logger.debug("%r resumes reading", self)
+        elif not self._paused:
+            if self._loop.get_debug():
+                logger.debug("%r not paused", self)
```

Comment:
I'd change these debug messages to more descriptive ones: 

* `%r not paused` -> `%r: ignoring resume_reading() call; already reading`
* `%r already paused` -> `%r: ignoring pause_reading() call; already paused`